pkg/podman/podman.go

```diff
@@ -0,0 +1,190 @@
+	if err := setupAPIListener(out); err != nil {
+		return nil, fmt.Errorf("creating new podman instance: %w", err)
+	}
+
+	conn, err := bindings.NewConnection(context.Background(), podmanSock)
+	if err != nil {
+		return nil, fmt.Errorf("creating new podman connection: %w", err)
+	}
+
+	return &Podman{
+		context: conn,
+		out:     out,
+	}, nil
+}
+
+// Import imports a tarball and saves it as a filesystem image
+//
+// Parameters:
+//   - tarball - path to the tarball to be imported
+//   - ref 	  - name for the image that will be created from the tarball
+func (p *Podman) Import(tarball, ref string) error {
+	zap.S().Infof("Importing image '%s' from tarball...", ref)
+	f, err := os.Open(tarball)
+	if err != nil {
+		return fmt.Errorf("opening tarball %s: %w", tarball, err)
+	}
+	_, err = images.Import(p.context, f, &images.ImportOptions{Reference: &ref})
+	if err != nil {
+		return fmt.Errorf("importing tarball %s: %w", tarball, err)
+	}
+
+	return nil
+}
+
+// Build looks for a 'Dockerfile' in the given context and build a podman image
+// from it.
+func (p *Podman) Build(imageContext, imageName string) error {
+	zap.S().Infof("Building image %s...", imageName)
+
+	logFile, err := os.Create(filepath.Join(p.out, podmanBuildLogFile))
+	if err != nil {
+		return fmt.Errorf("generating podman build log file: %w", err)
+	}
+	defer logFile.Close()
+
+	eOpts := entities.BuildOptions{
+		BuildOptions: define.BuildOptions{
+			ContextDirectory: imageContext,
+			Output:           imageName,
+			Out:              logFile,
+			Err:              logFile,
+		},
+	}
+
+	_, err = images.Build(p.context, []string{dockerfile}, eOpts)
+	if err != nil {
+		return fmt.Errorf("building image from context %s: %w", imageContext, err)
+	}
+
+	return nil
+}
+
+// Create creates a container from the given image. Returns the id of the container.
+func (p *Podman) Create(img string) (string, error) {
+	zap.S().Infof("Creating container from %s image...", img)
+
+	s := specgen.NewSpecGenerator(img, false)
+	createResponse, err := containers.CreateWithSpec(p.context, s, nil)
+	if err != nil {
+		return "", fmt.Errorf("creating container with spec %v: %w", s, err)
+	}
+
+	return createResponse.ID, nil
+}
+
+// Copy copies a file or directory from a source located in the container
+// to a destination located outside of the container.
+//
+// Note: No need to create a placeholder file/directory in dest. The file/directory will be created
+// automatically upon copying from source.
+func (p *Podman) Copy(id, src, dest string) error {
+	zap.S().Infof("Copying %s from container %s to %s", src, id, dest)
+
+	reader, writer := io.Pipe()
+	defer reader.Close()
+
+	copyFunc, err := containers.CopyToArchive(p.context, id, src, writer)
+	if err != nil {
+		return fmt.Errorf("creating copy function for archive from %s: %w", src, err)
+	}
+
+	go func() {
+		err := copyFunc()
+		if err != nil {
+			zap.S().Errorf("copying %w to %w: %w", src, dest, err)
```

Comment:
This is where we log the outcome so the %w formatter is not appropriate: 
```suggestion
			zap.S().Errorf("Copying %s to %s failed: %s", src, dest, err)
```